repose-aggregator/services/atom-feed/api/src/main/java/org/openrepose/nodeservice/atomfeed/AuthenticatedRequestFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.nodeservice.atomfeed;
+
+import java.net.URLConnection;
+
+/**
+ * A factory which will perform any necessary processing to authenticate a request to the Atom service.
+ */
+public interface AuthenticatedRequestFactory {
+
+    /**
+     * Mutates a request to adhere to some authentication scheme compatible with the Atom service.
+     *
+     * @param atomFeedUrlConnection The unauthenticated feed {@link URLConnection} with the Atom service.
+     * @param requestId             A unique identifier associated with all of the requests for this feed reading interaction.
+     * @return The {@link URLConnection} with authentication mutations applied.
+     */
+    URLConnection authenticateRequest(URLConnection atomFeedUrlConnection, String requestId, String reposeVersion);
```

Comment:
I hate these being on this method call. They are here for the admin call though aren't they, because we could put them on the main call ourselves. Instead of a pair of strings do we want to instead pas in a context object of some sort, that happens to have these strings? Then if we come up with other things we need later we can make additions to that class instead of having to change the method signature which is potentially backwards in compatible.